pkg/mflag/flag.go

```diff
@@ -1143,3 +1143,56 @@ func (f *FlagSet) Init(name string, errorHandling ErrorHandling) {
 	f.name = name
 	f.errorHandling = errorHandling
 }
+
+func (f *FlagSet) DeprecateAll() {
+	m := make(map[string]*Flag, len(f.formal))
+	for name, flag := range f.formal {
+		if name[0] != '#' {
+			m["#"+name] = flag
+		} else {
+			m[name] = flag
+		}
+	}
+	f.formal = m
+}
+
+type mergeVal struct {
+	Value
+	key  string
+	fset *FlagSet
+}
+
+func (v mergeVal) Set(s string) error {
+	return v.fset.Set(v.key, s)
+}
+
+func (v mergeVal) IsBoolFlag() bool {
+	if b, ok := v.Value.(boolFlag); ok {
+		return b.IsBoolFlag()
+	}
+	return false
+}
+
+func Merge(dest *FlagSet, flagsets ...*FlagSet) {
+	for _, fset := range flagsets {
+		for k, f := range fset.formal {
+			if _, ok := dest.formal[k]; ok {
+				var msg string
+				if fset.name == "" {
+					msg = fmt.Sprintf("flag redefined: %s", k)
+				} else {
+					msg = fmt.Sprintf("%s flag redefined: %s", fset.name, k)
+				}
+				fmt.Fprintln(fset.Out(), msg)
+				panic(msg) // Happens only if flags are declared with identical names
```

Comment:
fortunately for us, we don't need to keep the behavior backwards compatible, do we? :wink: